### PR TITLE
BUG: avoid OOB flow in read_gbq and to_gbq

### PR DESCRIPTION
**Pull request by tswast**
The console-based copy-paste-a-token flow has been deprecated. See:
https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html?m=1#disallowed-oob

- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1974,7 +1974,7 @@ def to_gbq(
         chunksize: int | None = None,
         reauth: bool = False,
         if_exists: str = "fail",
-        auth_local_webserver: bool = False,
+        auth_local_webserver: bool = True,
```
Comment (jreback) [inline pandas/core/frame.py:1977]: this is not a deprecation but just a change. You can do that but update the whatsnew note and put it in other API changes.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 1 above)
Comment (tswast) [reply pandas/core/frame.py:1981]: Done. Moved to "Other API changes"

File: doc/source/whatsnew/v1.5.0.rst [hunk 2]
```diff
@@ -196,7 +196,13 @@ See :ref:`install.dependencies` and :ref:`install.optional_dependencies` for mor
...
 Other API changes
 ^^^^^^^^^^^^^^^^^
--
+
+- BigQuery I/O methods :func:`read_gbq` and :meth:`DataFrame.to_gbq` default to
+  ``auth_local_webserver = True``. Google has deprecated the
+  ``auth_local_webserver = False`` `"out of band" (copy-paste) flow
+  <https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html?m=1#disallowed-oob>`_.
+  The ``auth_local_webserver = False`` option is planned to stop working in
+  October 2022.
```
Comment (jreback) [inline doc/source/whatsnew/v1.5.0.rst:205]: can you add this PR number (see other entries for how)

File: pandas/core/frame.py [hunk 3]
```diff
@@ -2016,7 +2016,7 @@ def to_gbq(
                 If table exists, drop it, recreate it, and insert data.
             ``'append'``
                 If table exists, insert data. Create if does not exist.
-        auth_local_webserver : bool, default False
+        auth_local_webserver : bool, default True
```
Comment (jreback) [inline pandas/core/frame.py:2019]: can you add versionchanged 1.5.0

File: pandas/io/gbq.py [hunk 4]
```diff
@@ -29,7 +29,7 @@ def read_gbq(
     index_col: str | None = None,
     col_order: list[str] | None = None,
     reauth: bool = False,
-    auth_local_webserver: bool = False,
+    auth_local_webserver: bool = True,
```
Comment (jreback) [inline pandas/io/gbq.py:32]: same as above

File: pandas/io/gbq.py [hunk 5]
```diff
@@ -63,7 +63,7 @@ def read_gbq(
     reauth : bool, default False
         Force Google BigQuery to re-authenticate the user. This is useful
         if multiple accounts are used.
-    auth_local_webserver : bool, default False
+    auth_local_webserver : bool, default True
```
Comment (jreback) [inline pandas/io/gbq.py:66]: same as above

File: pandas/io/gbq.py [hunk 6]
```diff
@@ -202,7 +202,7 @@ def to_gbq(
     chunksize: int | None = None,
     reauth: bool = False,
     if_exists: str = "fail",
-    auth_local_webserver: bool = False,
+    auth_local_webserver: bool = True,
```
Comment (jreback) [inline pandas/io/gbq.py:205]: same as above

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, some doc comments, pls merge master and ping on green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thank you @tswast !

**Pull request merged**